Bump to Java 11

JonasCir opened the pull request:
Bump the projects Java Version from 8 to 11.  

This also fixes issues with my IDE as parts of `JavaConverter` were already using Java 11 features.

cguz commented:
Jonas, 

This PR is not working with the new updates in the master branch, could you please check it with the new branch?

Please, consider the following:

- I am testing with the Java version 17. 
- The maven works. However, there are some tests that fail. When I have some time, I will check it.

JonasCir commented:
Hi @cguz ,
thanks for taking care of this in such a timely manner! :)

I rebased on `master` and resolved the conflicts. I also added a [GitHub Action](https://github.com/features/actions) to run CI tests, both for Java 11 and Java 17. GitHub Action is GitHub's own CI provider and it is free for public repositories. 

I could not reproduce any failing tests, so could you please pull the branch again and run `mvn verify`?
